File: Solutions/1605- Find Valid Matrix Given Row and Column Sums/Solution.java
class Solution {
    public int[][] restoreMatrix(int[] rowSum, int[] colSum) {
        int[][] answer = new int[rowSum.length][colSum.length];


        for (int i = 0;i < rowSum.length; i++) {
            for (int j = 0; j < colSum.length; j++) {
                int minNumber = Math.min(rowSum[i], colSum[j]);
                answer[i][j] = minNumber;
                rowSum[i] -= minNumber;
                colSum[j] -= minNumber;
            }
        }
        
        return answer;


    }
}

File: Solutions/1605- Find Valid Matrix Given Row and Column Sums/Solution.py
class Solution:
    def restoreMatrix(self, rowSum: List[int], colSum: List[int]) -> List[List[int]]:
        answer = list()
        for i in range(0,len(rowSum)):
            temp = list()
            for j in range(0, len(colSum)):
                min_numer = min(rowSum[i], colSum[j])
                temp.append(min_numer)
                rowSum[i] -= min_numer
                colSum[j] -= min_numer
            answer.append(temp)
        return answer
        